test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/ArrayBuilderTest.scala

```diff
@@ -190,6 +191,8 @@ class ArrayBuilderTest {
   }
 
   @Test def Unit_zeros_inline(): Unit = {
+    assumeFalse("Can't run on 2.12.0-M4.",
+        System.getProperty("scalac.hasBoxedUnitBug") != null || executingInJVM)
```

Comment:
It does not make sense to prevent `executingInJVM` here, and later do `if (!executingInJVM)`.

From the changes about `ArrayBuilder.ofUnit` in Scala 2.12.0-M4, I would suggest to remove the `else` part of the `if (!executingInJVM)` instead of using `|| executingInJVM` in the assumption.